storageClient/api.go

```diff
@@ -533,15 +533,14 @@ type InternalFetchClientMembership struct {
 	Capabilities []string  `json:"capabilities"`
 }
 
-// AllowedGroupsForPolicyRequest wraps the request object for finding the allowed reads from a group
+// AllowedGroupsForPolicyRequest wraps the request object for finding the allowed group reads for a list of content types
 type AllowedGroupsForPolicyRequest struct {
-	WriterID    string `json:"writer_id"`
-	ContentType string `json:"content_type"`
+	ContentTypes []string `json:"content_type"`
 }
 
-// AllowedGroupsForPolicyResponse wraps the response object for the group ids in allowed reads
+// AllowedGroupsForPolicyResponse wraps the response object for the allowed group reads for a list of content types
 type AllowedGroupsForPolicyResponse struct {
-	GroupIDs []uuid.UUID `json:"group_ids"`
+	GroupsSharedWith map[string][]uuid.UUID // A map of content type to list of groups that content type is shared with.
```

Comment:
you need to add the JSON tag `json:"group_ids"` for the SDK to unwrap the response